Common/Python/PythonData.cs

```diff
@@ -32,6 +32,21 @@ public class PythonData : DynamicData
         private readonly dynamic _supportedResolutions;
         private readonly dynamic _isSparseData;
         private readonly dynamic _requiresMapping;
+        private DateTime _endTime;
+
+        /// <summary>
+        /// The end time of this data. Some data covers spans (trade bars)
+        /// and as such we want to know the entire time span covered
+        /// </summary>
+        /// <remarks>
+        /// This property is overriden to allow different values for Time and EndTime
+        /// if they are set in the Reader. In the base implementation EndTime equals Time
+        /// </remarks>
+        public override DateTime EndTime
+        {
+            get => _endTime == default ? Time : _endTime;
+            set => _endTime = value;
```

Comment:
Is there a period? Does EndTime-Period = Time?

Reply:
If users just define `Time` and `EndTime` it will be simpler than asking them to define a third variable `Period`. But we could define a `Period` too.